accepted/2024/package-update-command.md

```diff
@@ -0,0 +1,286 @@
+NPM Audit sends the list of used packages to the package registry, which responds with which packages are vulnerable (and presumably what versions vulnerabilities are fixed with).
+
+There is an `npm update` command, to update all packages to the latest version.
+However, it appears to be constrained to the version ranges specified by each package in `package.json`, so in effect it just updates the lock file to the highest floating range requested.
+The command `npm install` is used to install packages, but the docs don't specify the behavior if a different version of the package is already installed.
+Neither command appears to have audit related functionality, or features to upgrade packages with known vulnerabilities, although audit runs by default on any install/update.
+
+### [cargo audit](https://github.com/RustSec/cargo-audit)
+
+Rust's cargo does not have audit functionality built in, and a crate must be installed, in addition to being run manually.
+The `cargo-audit` crate supports both checking for vulnerabilities, as well as a fix command.
+Cargo's audit appears to use a git repository as its data source.
+
+### [pip audit](https://pypi.org/project/pip-audit/)
+
+Python's pip does not appear to have an audit function built in, so it needs to be explicitly installed and run by customers who want to use it.
+It has a `--fix` command, but the documentation doesn't provide any additional information.
+The docs say that it gets advisory information from the PyPI JSON API, which appears to be a per-package details endpoint.
+
+- [dotnet outdated](https://github.com/dotnet-outdated/dotnet-outdated)
+
+The `dotnet-outdated` tool is a general package update tool for .NET projects on the CLI.
+It does not appear to have functionality related to updating only packages with security advisories.
+
+- [OWASP Dependency-Check (Maven)](https://owasp.org/www-project-dependency-check/)
+
+Maven is a build tool for Java, including package management.
+It supports plugins, and OWASP's Dependency-Check can be used as a plugin to detect packages with known vulnerabilities.
+The docs don't mention any tooling to automatically update package versions to without vulnerabilities.
+
+## Unresolved Questions
+
+<!-- What parts of the proposal do you expect to resolve before this gets accepted? -->
+<!-- What parts of the proposal need to be resolved before the proposal is stabilized? -->
+<!-- What related issues would you consider out of scope for this proposal but can be addressed in the future? -->
+
+- CPM integration
+
+CPM adds more complexity than other NuGet features.
+Should `dotnet package update` upgrade `GlobalPackageReference`s?
+Should it allow you to modify Directory.Packages.props when only one project is provided, not the whole solution?
+When run on a solution, should it check for unused `PackageVersion`s and provide a way to remove them?
```

Comment:
I'd say not yet, mostly cause of I dream of a command that can do that by itself later.